### PR TITLE
Demisto TAXII server

### PR DESCRIPTION
**Pull request by liorblob**
## Status
- [X] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/21747

## Description
Demisto TAXII server

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [X] 5.5.0

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+- display: Private Key (Required for HTTPS)
+  name: key
+  required: false
+  type: 14
+- additionalinfo: JSON string of users in a username - password
+  display: Users JSON
+  name: users
+  required: false
+  type: 14
+- additionalinfo: Ignores certificate and private key, and will run the EDL service
```
Comment (Itay4) [inline Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.yml]: are you sure its meant for that? i think this flag is meant to spin up both http and https server

File: Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    in HTTP
+  display: HTTP Server
+  hidden: false
+  name: http_flag
+  required: false
+  type: 8
+- additionalinfo: JSON string of indicator query collections.
+  defaultvalue: |-
+    {
+       "Spamhaus": "type:IP and reputation:BAD and source:Spamhaus Feed"
```
Comment (Itay4) [inline Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.yml]: why is this the default value?

File: Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py [hunk 3]
```diff
@@ -0,0 +1,938 @@
...
+        Args:
+            taxii_message: The collection request message.
+
+        Returns:
+            The collection management response.
+        """
+        taxii_feeds = [name for name, query in self.collections.items()]
+
+        if taxii_message.message_type != MSG_COLLECTION_INFORMATION_REQUEST:
+            raise ValueError('Invalid message, invalid Message Type')
```
[the remark this answers is not in the document]
Comment (liorblob) [reply Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py]: it means that the request is for collection information, but the message type is wrong.

File: Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py [hunk 4]
```diff
@@ -0,0 +1,938 @@
...
+
+        collection_info_response = CollectionInformationResponse(
+            generate_message_id(),
+            taxii_message.message_id
+        )
+
+        for feed in taxii_feeds:
+            collection_info = CollectionInformation(
+                feed,
```
[the remark this answers is not in the document]
Comment (liorblob) [reply Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py]: what do you mean?

File: Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py [hunk 5]
```diff
@@ -0,0 +1,938 @@
...
+
+            yield response
+
+            # yield the content blocks
+            indicator_query = self.collections[str(collection_name)]
+
+            for indicator in find_indicators_by_time_frame(indicator_query, exclusive_begin_time, inclusive_end_time):
+                try:
+                    json_stix = get_stix_indicator(indicator)
+                    demisto.info(str(json_stix))
```
[the remark this answers is not in the document]
Comment (liorblob) [reply Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py]: :|

File: Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py [hunk 6]
```diff
@@ -0,0 +1,938 @@
...
+            status=200,
+            headers={
+                'X-TAXII-Content-Type': 'urn:taxii.mitre.org:message:xml:1.1',
+                'X-TAXII-Protocol': 'urn:taxii.mitre.org:protocol:http:1.0'
+            },
+            mimetype='application/xml'
+        )
+
+
+SERVER: TAXIIServer
```
[the remark this answers is not in the document]
Comment (liorblob) [reply Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py]: we need the server in the flask functions

File: Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py [hunk 7]
```diff
@@ -0,0 +1,938 @@
...
+    Args:
+        uri: The namespace URI
+        name: The namespace name
+    """
+    # maec and cybox
+    namespace = mixbox.namespaces.Namespace(uri, name)
+    mixbox.idgen.set_id_namespace(namespace)
+
+
+def get_stix_indicator(indicator: dict) -> str:
```
[the remark this answers is not in the document]
Comment (liorblob) [reply Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py]: it's there

**Comment by kirbles19** [comment]
@liorblob Done.

**Pull request merged**